File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/TaskBaselines/taskbaselineschedule.rb
module Asposetasksjava
  module TaskBaselineSchedule
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
        
        onesec = 10000000
        onemin = 60 * onesec
        onehour = 60 * onemin
        
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new
		
    		# Creating TaskBaseline:
        task = project.getRootTask().getChildren().add("Task")
        project.setBaseline(Rjb::import('com.aspose.tasks.BaselineType').Baseline)

        baseline = task.getBaselines().toList().get(0)
        duration = baseline.getDuration().toDouble()
        baseline_duration = duration / onehour
        puts baseline_duration
        puts "Baseline Start: "  + baseline.getStart().to_string
        puts "Baseline Finish: " + baseline.getFinish().to_string
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTaskBaselines/TaskBaselineSchedule.php
<?php
namespace Aspose\Tasks\WorkingWithTaskBaselines;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\BaselineType as BaselineType;
class TaskBaselineSchedule{

    public static function run($dataDir=null){

        $onesec = 10000000;
        $onemin = 60 * $onesec;
        $onehour = 60 * $onemin;

        # Instantiate project object
        $project = new Project();

        # Creating TaskBaseline:
        $task = $project->getRootTask()->getChildren()->add("Task");
        $baselineType=new BaselineType();
        $project->setBaseline($baselineType->Baseline);

        $baseline = $task->getBaselines()->toList()->get(0);
        $duration = (double)(string)($baseline->getDuration());//.toDouble();
        $baseline_duration = $duration / $onehour;
        print $baseline_duration;
        print "Baseline Start: "  . (string)$baseline->getStart().PHP_EOL;//.to_string;
        print "Baseline Finish: " . (string)$baseline->getFinish().PHP_EOL;//.to_string;

    }

}
?>